src/MathOps.cpp

```diff
@@ -436,24 +436,45 @@ static void DoIReduce(Thread& th, BinaryOp* op)
 	UnaryOp* gUnaryOpPtr_##NAME = &gUnaryOp_##NAME; \
 	UNARY_OP_PRIM(NAME)
 #else
-#define DEFINE_UNOP_FLOATVV(NAME, CODE, VVNAME) \
-	struct UnaryOp_##NAME : public UnaryOp { \
-		virtual const char *Name() { return #NAME; } \
-		virtual double op(double a) { return CODE; } \
-		virtual void loopz(int n, const Z *x, int astride, Z *y) { \
-                        LOOP(i,n) { Z a = *x; y[i] = CODE; x += astride; }  \
-		} \
-	}; \
-	UnaryOp_##NAME gUnaryOp_##NAME; \
-	UnaryOp* gUnaryOpPtr_##NAME = &gUnaryOp_##NAME; \
-	UNARY_OP_PRIM(NAME)
+#if SAMPLE_IS_DOUBLE
+	typedef Eigen::Map<Eigen::ArrayXd, 0, Eigen::InnerStride<>> ZArr;
+#else
+	typedef Eigen::Map<Eigen::ArrayXf, 0, Eigen::InnerStride<>> ZArr;
+#endif
 
-#define DEFINE_UNOP_FLOATVV2(NAME, CODE, VVCODE) \
+ZArr zarr(const Z *vec, int n, int stride) {
+	#if SAMPLE_IS_DOUBLE
+		return ZArr((double *)vec, n, Eigen::InnerStride<>(stride));
+	#else
+		return ZArr((float *)vec, n, Eigen::InnerStride<>(stride));
+	#endif
```

Comment:
this could be simplified to `return ZArr((Z *)vec, n, Eigen::InnerStride<>(stride));`, no?